src/server.mjs

```diff
@@ -1446,16 +1494,12 @@ async function getBedRegions(bed) {
       
       // Work out where it should be locally.
       const chunk_path = bedChunkLocalPath(bed, chunk);
-      // download the tracks file if bed is an url
-      if (isURL) {
-        await retrieveChunk(bed, chunk, false);
-      }
 
+      // See if we have downloaded tracks.json in a previous instance
       let track_json = path.resolve(chunk_path, "tracks.json");
 
-      let tracks_array = [];
-
-      // If json file specifying the tracks exists
+      // If json file specifying the tracks exists, pass its information into a tracks object
+      // future selection of this region won't re-fetch tracks.json
       if (fs.existsSync(track_json)) {
```

Comment:
We might want to think about not sending any track data with the `getBedRegions` response (in order to simplify things and reduce the number of distinct ways you have to worry about sending track data).

Or maybe only sending track data when there are just a few regions?

If somebody makes a *really* big BED, and mounts it onto the server, we don't want to have to be jumping all over the disk opening thousands of JSON files to answer the request, especially if the user is only actually going to need a few of them, one at a time.